test/e2e/package.json

```diff
@@ -16,7 +16,8 @@
 	},
 	"scripts": {
 		"encryptconfig": "openssl enc -md sha1 -aes-256-cbc -pass env:CONFIG_KEY -out ./config/encrypted.enc -in ./config/local-$NODE_CONFIG_ENV.json",
-		"decryptconfig": "openssl enc -md sha1 -aes-256-cbc -d -pass env:CONFIG_KEY -in ./config/encrypted.enc -out ./config/local-$NODE_CONFIG_ENV.json"
+		"decryptconfig": "openssl enc -md sha1 -aes-256-cbc -d -pass env:CONFIG_KEY -in ./config/encrypted.enc -out ./config/local-$NODE_CONFIG_ENV.json",
+		"checktypes": "tsc --project ./tsconfig.json"
```

Comment:
See https://github.com/Automattic/wp-calypso/pull/54734#issuecomment-884637006.

I think we should drop this, and instead change the TeamCity build to also include `test/e2e/packages.json` when typechecking the repo for now.

Then we need to decide how to provide a script to typecheck the whole repo, the same way we have scripts to lint the whole repo or run all unit tests (pretty much the existing `yarn build-packages` already does it)

Reply:
Makes sense - I again think I was viewing these subdirectories/packages too atomically!

So, I was feeling exploratory this morning, and just went for giving `ts-jest` a whirl as the way of fixing this. I think especially for our e2e tests, having type checking at the time of running tests is going to be especially useful because people use `yarn jest <their specific script>` so much when working on individual spec, and "failing fast" with type errors there sounds so useful. This method also means that we don't have to add any new commands or worry about special handling in teamcity for these tests!

Also, `ts-jest` now supports passing in a babel config to use, so it ends up using the configuration under the hood as the old way!

Anyway, I figured we could give it a whirl here, and if we like it maybe we can look into that for unit tests in the future too?

Reply:
> and "failing fast" with type errors there sounds so useful.

I'd argue that the editor already does that. Having the types checked when I save the file, and then when I run it seems redundant. 

> worry about special handling in teamcity for these tests!

I still think we should change the TC script to typecheck tests as well. Otherwise these tests do get a special handling, as the rest of the repo is typechecked in that build except `test/e2e`.

-- 

What's the performance impact of typechecking tests when running them (i.e. how slower is CI with this setup?). Is it noticeable when running them locally? I think those are the only "real" concerns I have, everything else boils down to personal preferences.

Reply:
>I still think we should change the TC script to typecheck tests as well. Otherwise these tests do get a special handling, as the rest of the repo is typechecked in that build except test/e2e.

That makes sense. I saw your other comment too about `ts-jest` and its bloat, and that's a definite real concern. I've read online too that people sometimes complain about `ts-jest`s overall performance, and that seems to be a critical piece here.

I'll switch back to just using Babel for the time being! And I'll defer any type-checking for the TC build in CI.